File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int ans =nums.size() ;
        int lo = 0,hi = nums.size()-1;
        while(lo<=hi){
            int mid = lo + (hi-lo)/2;
            if(nums[mid]==target) return mid;
            else if (nums[mid]>target){
                ans = mid;
                hi = mid-1;
            }else{
                lo = mid+1;
            }
        }
        return ans;
    }
};

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        ans = len(nums)
        lo, hi = 0,len(nums)-1
        while lo <= hi:
            mid = lo + (hi-lo)//2
            if nums[mid]==target:
                return mid
            elif nums[mid] > target:
                ans = mid
                hi = mid-1
            else :
                lo = mid+1
        return ans